### ISSUE TITLE
What is expected behavior for database sync when tables are removed?

### ISSUE THREAD

**Issue by orware** [metabase/metabase#5115]
### Bugs

I had a MySQL (MemSQL actually) database that I had added to Metabase the other week and initially when I added it in to Metabase it had a number of tables I had already copied over.

However, since I was experimenting with MemSQL I've since destroyed and recreated the database and currently haven't brought over the same tables/data just yet. I would have expected the syncing process to automatically update Metabase so that tables that may no longer exist be automatically removed (e.g. right now they show up still when going to add in a new question, even if I open up a fresh browser incognito window). Right now it seems like it is just keeping the existing tables there and not disabling/hiding them automatically.

I could delete that particular database and then add it again fresh which I think will fix things since the new database will have a new ID and will start off blank and just bring in the basics of what is there in the newly created database, but I just wanted to share the observation since it doesn't feel like the expected behavior I would want as a user.

-------

**Comment by salsakran** [comment]
We mark tables + fields as "inactive" if they disappear. If they re-appear, we mark them as active again. This is to allow things like ETL created tables to retain their attached questions.

By default, we sync once a day at midnight server time. 

If a sync can't connect to a database, it doesn't do anything with regards to marking tables as active or inactive.

**Comment by orware** [comment]
@salsakran, 

Hi Sameer, what you've described sounds like it would have resulted in the correct behavior...however Metabase was up over the whole weekending and this set of changes occurred on Friday so it should have inactivated those tables already that no longer existed in the database...yet they were still there when I tried earlier today.

**Comment by salsakran** [comment]
what version are you on?

**Comment by orware** [comment]
@salsakran I'm on an in-between version of .24 (a not quite final .24 containing the LDAP additions from a few weeks ago) since I replaced my .23 version with an in-progress JAR file containing the LDAP additions being worked on here in another issue on the tracker by William.

**Comment by salsakran** [comment]
it was fixed in https://github.com/metabase/metabase/issues/3331
any deviation from that behaviour is a bug. Can you reproduce this with an official version? We don't take bugs filed against (or otherwise support) self-rolled builds.

**Comment by orware** [comment]
@salsakran I think his snapshot would have occurred after that one was committed so it should have technically had that fix within it already:

This is the snapshot William built his JAR file off of (from about 17 days ago, and #3331 was committed at the beginning of March):
https://github.com/wwwiiilll/metabase/tree/ldap-auth-snapshot

**Comment by orware** [comment]
I've asked if he can rebuild his JAR file using the latest .24.1 version so I can bring in some of the officially released new features but I'm not sure when he might be able to generate the new JAR file (that way I can continue using the LDAP functionality).

**Comment by VikramTiwari** [comment]
@salsakran Not sure if totally related but I have seen this happen:
- A table contains n fields
- table is synced and hence fields are created
- table is deleted
- on sync again, fields are marked as inactive
- table is readded, same table name, n - x fields
- [ISSUE]: This causes a complete resync and creation on new fields altogether rather than using the existing fields
- So in the end, fields table contains duplicate values wtih different metabase_table references but same raw_table references
- The reason for this is because a totally new table reference is created in metabase_table while raw_table value is same
- Not sure what is wrong in letting it use the existing reference between the two and mark columns as private.

Example:
raw_table: `id: 1`
metabase_table: `id: 1`
synced fields:
```
id: 1, name: a
id: 2, name: b
id: 3, name: c
```
(remove table, re-add with `c` column removed, sync)
raw_table: `id: 1`
metabase_table: `id: 2`
synced_fields:
```
id: 1, name: a, active: false
id: 2, name: b, active: false
id: 3, name: c, active: false
id: 4, name: a, active: true
id: 5, name: b, active: true
```

**Comment by flamber** [comment]
Closing, since it seemed like it was fixed by #3331. Related to #5952
